### ISSUE TITLE
Unable to checkout 

### ISSUE THREAD

**Issue by spignataro** [magento/magento2#34348]
### Preconditions (*)

1. Magento 2.4.3-p1

### Steps to reproduce (*)

1. Upgrade from 2.4.3 to 2.4.3-p1
2. This was done through composer
3. Go to the checkout screen and fill out all of the required fields. When you click to go and checkout the payment rest API 500 errors and logs a error in the system.log file.

### Expected result (*)

1. When clicking on the checkout button we should go to the receipt screen.

### Actual result (*)

1. System errors out and logs to the system.log file:
```
[2021-10-15 01:25:43] main.CRITICAL: Error: Call to a member function addItem() on array in /home/staging/public_html/vendor/magento/module-sales/Model/Order/Invoice.php:552
Stack trace:
#0 /home/staging/public_html/vendor/magento/module-sales/Model/Service/InvoiceService.php(173): Magento\Sales\Model\Order\Invoice->addItem(Object(Magento\Sales\Model\Order\Invoice\Item))
#1 /home/staging/public_html/vendor/magento/module-sales/Model/Order.php(2139): Magento\Sales\Model\Service\InvoiceService->prepareInvoice(Object(Magento\Sales\Model\Order\Interceptor), Array)
#2 /home/staging/public_html/vendor/magento/module-sales/Model/Order/Payment/Operations/AbstractOperation.php(69): Magento\Sales\Model\Order->prepareInvoice()
#3 /home/staging/public_html/vendor/magento/module-sales/Model/Order/Payment/Operations/CaptureOperation.php(65): Magento\Sales\Model\Order\Payment\Operations\AbstractOperation->invoice(Object(Magento\Sales\Model\Order\Payment\Interceptor))
#4 /home/staging/public_html/vendor/magento/module-sales/Model/Order/Payment/Processor.php(85): Magento\Sales\Model\Order\Payment\Operations\CaptureOperation->capture(Object(Magento\Sales\Model\Order\Payment\Interceptor), NULL)
#5 /home/staging/public_html/vendor/magento/module-sales/Model/Order/Payment.php(488): Magento\Sales\Model\Order\Payment\Processor->capture(Object(Magento\Sales\Model\Order\Payment\Interceptor), NULL)
#6 /home/staging/public_html/vendor/magento/module-sales/Model/Order/Payment.php(468): Magento\Sales\Model\Order\Payment->capture(NULL)
#7 /home/staging/public_html/vendor/magento/module-sales/Model/Order/Payment.php(385): Magento\Sales\Model\Order\Payment->processAction('authorize_captu...', Object(Magento\Sales\Model\Order\Interceptor))
#8 /home/staging/public_html/vendor/magento/module-sales/Model/Order.php(992): Magento\Sales\Model\Order\Payment->place()
#9 /home/staging/public_html/vendor/magento/module-sales/Model/Order.php(1214): Magento\Sales\Model\Order->_placePayment()
#10 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Sales\Model\Order->place()
#11 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Sales\Model\Order\Interceptor->___callParent('place', Array)
#12 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Sales\Model\Order\Interceptor->Magento\Framework\Interception\{closure}()
#13 /home/staging/public_html/generated/code/Magento/Sales/Model/Order/Interceptor.php(32): Magento\Sales\Model\Order\Interceptor->___callPlugins('place', Array, Array)
#14 /home/staging/public_html/vendor/magento/module-sales/Model/Service/OrderService.php(209): Magento\Sales\Model\Order\Interceptor->place()
#15 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Sales\Model\Service\OrderService->place(Object(Magento\Sales\Model\Order\Interceptor))
#16 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Sales\Model\Service\OrderService\Interceptor->___callParent('place', Array)
#17 /home/staging/public_html/vendor/magento/module-inventory-sales/Plugin/Sales/OrderManagement/AppendReservationsAfterOrderPlacementPlugin.php(195): Magento\Sales\Model\Service\OrderService\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Sales\Model\Order\Interceptor))
#18 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(135): Magento\InventorySales\Plugin\Sales\OrderManagement\AppendReservationsAfterOrderPlacementPlugin->aroundPlace(Object(Magento\Sales\Model\Service\OrderService\Interceptor), Object(Closure), Object(Magento\Sales\Model\Order\Interceptor))
#19 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Sales\Model\Service\OrderService\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Sales\Model\Order\Interceptor))
#20 /home/staging/public_html/generated/code/Magento/Sales/Model/Service/OrderService/Interceptor.php(32): Magento\Sales\Model\Service\OrderService\Interceptor->___callPlugins('place', Array, Array)
#21 /home/staging/public_html/vendor/magento/module-quote/Model/QuoteManagement.php(575): Magento\Sales\Model\Service\OrderService\Interceptor->place(Object(Magento\Sales\Model\Order\Interceptor))
#22 /home/staging/public_html/vendor/magento/module-quote/Model/QuoteManagement.php(456): Magento\Quote\Model\QuoteManagement->submitQuote(Object(Magento\Quote\Model\Quote\Interceptor), Array)
#23 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Quote\Model\QuoteManagement->submit(Object(Magento\Quote\Model\Quote\Interceptor))
#24 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Quote\Model\QuoteManagement\Interceptor->___callParent('submit', Array)
#25 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Quote\Model\QuoteManagement\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Quote\Model\Quote\Interceptor))
#26 /home/staging/public_html/generated/code/Magento/Quote/Model/QuoteManagement/Interceptor.php(32): Magento\Quote\Model\QuoteManagement\Interceptor->___callPlugins('submit', Array, Array)
#27 /home/staging/public_html/vendor/magento/module-quote/Model/QuoteManagement.php(414): Magento\Quote\Model\QuoteManagement\Interceptor->submit(Object(Magento\Quote\Model\Quote\Interceptor))
#28 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Quote\Model\QuoteManagement->placeOrder(1042)
#29 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Quote\Model\QuoteManagement\Interceptor->___callParent('placeOrder', Array)
#30 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Quote\Model\QuoteManagement\Interceptor->Magento\Framework\Interception\{closure}(1042)
#31 /home/staging/public_html/generated/code/Magento/Quote/Model/QuoteManagement/Interceptor.php(23): Magento\Quote\Model\QuoteManagement\Interceptor->___callPlugins('placeOrder', Array, Array)
#32 /home/staging/public_html/vendor/magento/module-checkout/Model/PaymentInformationManagement.php(125): Magento\Quote\Model\QuoteManagement\Interceptor->placeOrder(1042)
#33 [internal function]: Magento\Checkout\Model\PaymentInformationManagement->savePaymentInformationAndPlaceOrder(1042, Object(Magento\Quote\Model\Quote\Payment), Object(Magento\Quote\Model\Quote\Address\Interceptor))
#34 /home/staging/public_html/vendor/magento/module-webapi/Controller/Rest/SynchronousRequestProcessor.php(95): call_user_func_array(Array, Array)
#35 /home/staging/public_html/vendor/magento/module-webapi/Controller/Rest.php(188): Magento\Webapi\Controller\Rest\SynchronousRequestProcessor->process(Object(Magento\Framework\Webapi\Rest\Request\Proxy))
#36 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http))
#37 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Webapi\Controller\Rest\Interceptor->___callParent('dispatch', Array)
#38 /home/staging/public_html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Webapi\Controller\Rest\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#39 /home/staging/public_html/generated/code/Magento/Webapi/Controller/Rest/Interceptor.php(23): Magento\Webapi\Controller\Rest\Interceptor->___callPlugins('dispatch', Array, Array)
#40 /home/staging/public_html/vendor/magento/framework/App/Http.php(116): Magento\Webapi\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#41 /home/staging/public_html/vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http->launch()
#42 /home/staging/public_html/pub/index.php(29): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor))
#43 {main} [] []
```
2. No other results occur. We are using Braintree as our payment gateway. Nothing has changed in the settings.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [x] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by spignataro** [comment]
> Can confirm... We're dealing with same issue currently.

Are you using Braintree as your payment processor? We are wondering if it has to do with the processor as we just tested and everything works with cash on delivery.

**Comment by spignataro** [comment]
After testing - I can confirm it is due to the Mirasvit modules. I just deactivated them and bam - everything worked.

**Comment by spignataro** [comment]
I am going to assume that the issue stems from 1.0.20 which was updated a few days back.

**Comment by KZNcode** [comment]
Yup... Tested and verified functional...


************************************

*****

Roman Zhalybin (Customer Care Team)
Oct 18, 2021 12:06 PM
Hello,

Sorry for the trouble you have.

We have fixed this issue and released the new 1.0.21 version of the Order Management extension. Please update the extension.

best regards,
Roman
Bad
Ok
Great
Maria Antoshkiv (Customer Care Team)
Oct 18, 2021 10:26 AM
Hi there,

Sorry for the delayed reply, our office was closed on the weekend.

As for compatibility - our dev will fix this issue as soon as possible and notify you in this ticket.

As for types of tickets in our system - thank you for pointing attention to such misunderstanding. All feedback is highly appreciated. We'll take your request into consideration.

Thank you.

best regards,
Maria
Bad
Ok
Great
Ales Ales
Oct 16, 2021 12:16 AM
see: https://github.com/magento/magento2/issues/34348

Also, why is "Technical Support" option under this ticketing system not allowed? It's saying support has expired (we bought this in July 2021) ... Your extensions should be Magento version compatible across updates...  And this is a "Technical Issue" .... Hence, add another option for something like "paid technical support" and leave "Technical Support" open to everyone.
